### ISSUE TITLE
tikv-ctl failed to connect to tikv when TLS is enabled

### ISSUE THREAD

**Issue by weekface** [tikv/tikv#7891]
### What version of TiKV are you using?

```
TiKV
Release Version:   4.0.0-rc.2
Edition:           Community
Git Commit Hash:   2fdb2804bf8ffaab4b18c4996970e19906296497
Git Commit Branch: heads/refs/tags/v4.0.0-rc.2
UTC Build Time:    2020-05-15 11:58:49
Rust Version:      rustc 1.42.0-nightly (0de96d37f 2019-12-19)
Enable Features:   jemalloc portable sse protobuf-codec
Profile:           dist_release
```

### What operating system and CPU are you using?

```
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 142
model name	: Intel(R) Core(TM) i5-7Y54 CPU @ 1.20GHz
stepping	: 9
cpu MHz		: 1300.000
cache size	: 4096 KB
physical id	: 0
siblings	: 1
core id		: 0
cpu cores	: 1
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht pbe syscall nx pdpe1gb lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq dtes64 ds_cpl ssse3 sdbg fma cx16 xtpr pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch pti fsgsbase bmi1 avx2 bmi2 erms xsaveopt arat
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs
bogomips	: 2592.00
clflush size	: 64
cache_alignment	: 64
address sizes	: 39 bits physical, 48 bits virtual
power management:

processor	: 1
vendor_id	: GenuineIntel
cpu family	: 6
model		: 142
model name	: Intel(R) Core(TM) i5-7Y54 CPU @ 1.20GHz
stepping	: 9
cpu MHz		: 1300.000
cache size	: 4096 KB
physical id	: 1
siblings	: 1
core id		: 0
cpu cores	: 1
apicid		: 1
initial apicid	: 1
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht pbe syscall nx pdpe1gb lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq dtes64 ds_cpl ssse3 sdbg fma cx16 xtpr pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch pti fsgsbase bmi1 avx2 bmi2 erms xsaveopt arat
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs
bogomips	: 2589.75
clflush size	: 64
cache_alignment	: 64
address sizes	: 39 bits physical, 48 bits virtual
power management:

processor	: 2
vendor_id	: GenuineIntel
cpu family	: 6
model		: 142
model name	: Intel(R) Core(TM) i5-7Y54 CPU @ 1.20GHz
stepping	: 9
cpu MHz		: 1300.000
cache size	: 4096 KB
physical id	: 2
siblings	: 1
core id		: 0
cpu cores	: 1
apicid		: 2
initial apicid	: 2
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht pbe syscall nx pdpe1gb lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq dtes64 ds_cpl ssse3 sdbg fma cx16 xtpr pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch pti fsgsbase bmi1 avx2 bmi2 erms xsaveopt arat
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs
bogomips	: 2591.52
clflush size	: 64
cache_alignment	: 64
address sizes	: 39 bits physical, 48 bits virtual
power management:

processor	: 3
vendor_id	: GenuineIntel
cpu family	: 6
model		: 142
model name	: Intel(R) Core(TM) i5-7Y54 CPU @ 1.20GHz
stepping	: 9
cpu MHz		: 1300.000
cache size	: 4096 KB
physical id	: 3
siblings	: 1
core id		: 0
cpu cores	: 1
apicid		: 3
initial apicid	: 3
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht pbe syscall nx pdpe1gb lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq dtes64 ds_cpl ssse3 sdbg fma cx16 xtpr pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch pti fsgsbase bmi1 avx2 bmi2 erms xsaveopt arat
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs
bogomips	: 2631.58
clflush size	: 64
cache_alignment	: 64
address sizes	: 39 bits physical, 48 bits virtual
power management:
```

### Steps to reproduce

In [tidb-operator](https://pingcap.com/docs/tidb-in-kubernetes/stable/tidb-operator-overview/), I started the TiDB Cluster with TLS enabled: https://pingcap.com/docs/tidb-in-kubernetes/stable/enable-tls-between-components/.

The TiKV Server certificate:

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            8d:ed:d2:77:d2:ab:d4:72:63:24:96:68:40:5d:30:a1
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: O = cert-manager, CN = TiDB
        Validity
            Not Before: May 21 03:18:45 2020 GMT
            Not After : May 21 03:18:45 2021 GMT
        Subject: O = PingCAP, CN = TiDB
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:ac:6b:47:77:28:79:12:98:10:94:a3:84:b3:0e:
                    c9:59:25:c9:ac:15:56:6d:30:1a:10:89:3f:c7:11:
                    2e:b1:1d:ed:05:38:1b:82:dc:cf:d9:13:d5:62:63:
                    53:a7:20:5f:e3:38:70:96:a5:54:31:2c:4f:a0:0c:
                    2f:f7:27:64:18:83:a6:30:dd:c8:ab:db:73:19:e1:
                    fa:7b:70:1b:e5:b9:ac:42:0f:53:61:c0:f7:d6:5a:
                    1c:32:fc:75:db:27:7b:2a:b7:e9:53:4a:59:30:37:
                    b4:04:25:92:8d:b1:f4:26:7e:31:96:6a:4f:5a:47:
                    2c:8a:62:50:25:03:13:da:0c:07:9a:08:c0:5f:d3:
                    58:e3:77:26:f1:7e:75:52:71:89:7c:8a:40:9c:ef:
                    bc:9b:da:a4:c1:f7:7d:70:39:db:dc:65:65:99:47:
                    c8:d6:35:35:ee:68:d2:03:f2:58:33:a9:54:58:7a:
                    b2:c4:c8:97:47:3a:8a:f0:c1:b4:bb:c9:24:5d:81:
                    d4:16:08:73:86:ed:c8:96:fd:16:fb:48:09:95:36:
                    32:19:56:e9:72:70:80:99:e1:4e:cb:a1:4a:40:22:
                    64:2b:4d:46:a8:f9:06:91:50:a8:85:2c:0e:73:7a:
                    75:22:c1:4d:59:7d:13:68:ab:3b:94:52:ff:18:71:
                    d6:6b
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Alternative Name:
                DNS:basic-tikv, DNS:basic-tikv.default, DNS:basic-tikv.default.svc, DNS:basic-tikv-peer, DNS:basic-tikv-peer.default, DNS:basic-tikv-peer.default.svc, DNS:*.basic-tikv-peer, DNS:*.basic-tikv-peer.default, DNS:*.basic-tikv-peer.default.svc, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1
    Signature Algorithm: sha256WithRSAEncryption
         d7:5e:93:eb:51:56:ea:cf:0b:cb:71:16:c4:9d:52:25:80:a0:
         57:04:a6:20:f1:94:23:60:20:36:cf:cf:cc:3f:4b:a8:06:b2:
         78:6d:8a:17:18:0d:26:a5:be:68:02:ae:02:bd:f6:50:c8:94:
         50:65:1b:f0:40:93:36:25:d5:f1:1e:8b:f3:99:57:8b:1c:5a:
         1e:3e:11:c2:17:e8:8a:61:3d:5b:af:74:f6:04:b1:a0:b9:82:
         60:b4:f5:36:fc:3c:e8:ba:f9:5e:6b:12:3d:94:8e:0a:9f:69:
         e7:81:c6:22:01:8b:d5:19:31:04:9b:d9:34:99:2a:f7:a6:3e:
         bb:1f:9a:46:d2:6a:fb:a7:dd:8d:1c:92:d2:11:1c:f6:63:04:
         c6:91:05:a4:cc:82:2a:bc:f3:fd:e4:02:aa:0c:c1:09:2d:ee:
         9d:71:9a:17:ff:01:5a:3b:d4:a5:7c:8b:58:24:b2:01:61:79:
         45:76:16:8c:8e:82:70:f4:fa:70:b7:c3:9e:5b:7a:9e:d0:3e:
         21:7c:d6:90:ba:12:16:eb:b3:d4:6c:8d:00:2b:1b:86:9d:40:
         ca:c7:27:6c:b3:06:8a:ca:71:af:d5:23:a2:a2:d2:38:0a:dc:
         78:66:48:01:37:eb:0d:c2:61:30:3f:2e:33:ec:00:25:a6:34:
         b7:ad:d6:0a
```

I added DNS *.basic-tikv-peer.default to TiKV certificate's SAN. Then i use tikv-ctl and --host=basic-tikv-0.basic-tikv-peer.default:20160 to connect to TiKV Server:

```
$ ./tikv-ctl --ca-path ca --cert-path cert --key-path key --host basic-tikv-0.basic-tikv-peer.default:20160 cluster
```

### What did you expect?

it should succeed.

### What did happened?

It failed:

```
E0521 06:21:05.640890000    4567 ssl_transport_security.cc:1246] Handshake failed with fatal error SSL_ERROR_SSL: error:1408F10B:SSL routines:ssl3_get_record:wrong version number.
DebugClient::get_cluster_info: RpcFailure: 14-UNAVAILABLE failed to connect to all addresses
```

**Comment by weekface** [comment]
@hunterlxt PTAL

**Comment by hunterlxt** [comment]
I have tested by creating cluster with 1PD 1TiKV 1TiDB with nightly and v4.0.0.0-rc.2 binary files. And tikv-ctl worked fine when tls is enabled, @weekface and I switched certs and we found my cluster work correctly again but @weekface's reported same error again.

After simple analysis, I think this is difference brought by the way of establishing cluster.

This issue will be still open until the problem be found.
